### PR TITLE
create progress bar for books

### PR DESCRIPTION
**Pull request by b0b3rt**
Adds a progress bar for "books".  Each "box" in the progress bar is a link + tooltip with the usual post preview.

Also adds some type-safety for checking AB test group assignment.

OLD

NEW

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1202577453515220) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/lib/abTests.ts [hunk 1]
```diff
@@ -33,3 +33,19 @@ export const collectionsPageABTest = new ABTest({
...
+    control: {
+      description: "Original BooksItem without the progress bar",
+      weight: 1
+    },
+    progressBar: {
+      description: "Progress bar enabled",
+      weight: 1
+    }
+  }
+})
```
Comment (jpaddison3) [inline packages/lesswrong/lib/abTests.ts:51]: [Nitpick] [Files should end with newlines](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline). That one of the few autoformats we haven't turned off here on ForumMagnum, and so will cause spurious diff if left in.

Comment (b0b3rt) [reply packages/lesswrong/lib/abTests.ts]: Done, thanks

**Comment by jpaddison3** [comment]
Maybe allow the collection author to disable it? For sequences you want to have a "yeah let's get through it!" But maybe you're doing something where it makes less sense? IDK, I think we'd probably enable it on Intro Curriculum, so ¯\\\_(ツ)_/¯

File: packages/lesswrong/components/sequences/BooksProgressBar.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import React from 'react';
+import { Link } from '../../lib/reactRouterWrapper';
+import { postGetPageUrl } from '../../lib/collections/posts/helpers';
+import classNames from 'classnames';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    display: "inline"
+  },
+  post: {
```
Comment (darkruby501) [inline packages/lesswrong/components/sequences/BooksProgressBar.tsx:11]: rename to something clearer about what is this

**Comment by b0b3rt** [comment]
Probably a good idea to follow up on; would also think about whether to let users themselves disable it (rather than authors, or maybe even in addition to).

File: packages/lesswrong/components/sequences/BooksProgressBar.tsx (hunk 2 above)
Comment (b0b3rt) [reply packages/lesswrong/components/sequences/BooksProgressBar.tsx]: done

**Comment by Raemon** [comment]
> Maybe allow the collection author to disable it? For sequences you want to have a "yeah let's get through it!" But maybe you're doing something where it makes less sense? IDK, I think we'd probably enable it on Intro Curriculum, so ¯\_(ツ)_/¯

Note that collections are only made by admins. This decision wouldn't come up very often. (If you end up having a collection where you don't want to encourage people to read through as hard, does seem fine to make a parameter that turns off the progress bar).

**Review by darkruby501 — APPROVED** [review]

**Pull request merged**